APIKEY MODEL

jojothomas1515 opened the pull request:
if any error tell me

jojothomas1515 committed:
api table

jojothomas1515 committed:
api table

jojothomas1515 committed:
add the __repr__ function

jojothomas1515 committed:
added the tablename argument

jojothomas1515 committed:
added the tablename argument

a11rew committed:
Merge branch 'master' of https://github.com/a11rew/vip-service-backend into jojo-api-key

a11rew committed:
refactor: formatting

a11rew commented:
Hi Jojo, 

see https://dbdiagram.io/d/636cb38ac9abfc6111719f7d:

The `apikey` table should have a user column with a foreign key relationship to the `user` table, you already have the key column so you'd just need to add that.

Also, I'm not sure this implementation of the API key generator guarantees uniqueness. i.e. different rows could have the same API key. Correct me if I'm wrong. Uniqueness is a requirement since lookups will be done using the key.   

You could implement it yourself or use the already created UUID model in addition to a unique constraint

jojothomas1515 committed:
changed

jojothomas1515 commented:
following the diagram i removed it as the parent to history  since the users table is the parent, and for the random a api key generator , i notice when i check the uuid model that even the random uuid generator has not been implement , it just a place holder that is there so i left  mine the way it was , if you  still need me to change it, then no problem , but i think this one is okay for presentation 


and also i added the unique constraint to it so the database will not allow duplicate fields with the same apikey

jojothomas1515 committed:
added apikey constraint

a11rew commented on app/db/models/api.py:
```diff
@@ -0,0 +1,33 @@
+    __table_args__ = (UniqueConstraint("apikey"),)
+    apikey: str = Field(default=get_random_string(30), index=True)
+    
+    user_id: Optional[str] = Field(default=None, foriegn_key="user.id")
```
This shouldn't be optional

jojothomas1515 committed:
optional removed
